src/scripts/main.js

```diff
@@ -1,3 +1,202 @@
 'use strict';
 
-// write code here
+const trList = Array.from(document.querySelectorAll('tr th')).slice(0, 5);
+const tBody = document.querySelector('tbody');
+const rows = Array.from(tBody.querySelectorAll('tr'));
+
+function parseSalary(salaryString) {
+  return Number(salaryString.replace(/[$,]/g, ''));
+}
+
+const sortState = [];
+
+trList.forEach((headerTh, columnIndex) => {
+  headerTh.addEventListener('click', () => {
+    const ascSort = sortState[columnIndex];
+
+    sortTable(columnIndex, ascSort);
+
+    sortState.fill(false);
+    sortState[columnIndex] = !ascSort;
+  });
+});
+
+function sortTable(columnIndex, ascSort) {
+  const sortedRows = rows.sort((a, b) => {
+    const aText = a.children[columnIndex].textContent.trim();
+    const bText = b.children[columnIndex].textContent.trim();
+
+    const aNum = parseSalary(aText);
+    const bNum = parseSalary(bText);
+
+    if (!isNaN(aNum) && !isNaN(bNum)) {
+      if (ascSort) {
+        return bNum - aNum;
+      }
+
+      return aNum - bNum;
+    } else {
+      if (ascSort) {
+        return bText.localeCompare(aText);
+      }
+
+      return aText.localeCompare(bText);
+    }
+  });
+
+  sortedRows.forEach((row) => tBody.appendChild(row));
+}
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    if (row.classList.contains('active')) {
+      row.classList.remove('active');
+    } else {
+      rows.forEach((r) => r.classList.remove('active'));
+      row.classList.add('active');
+    }
+  });
+});
+
+const form = document.createElement('form');
+const button = document.createElement('button');
+
+const attributes = ['name', 'position', 'office', 'age', 'salary'];
+
+const options = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+for (let i = 0; i <= 4; i++) {
+  const input = document.createElement('input');
+  const label = document.createElement('label');
+  const select = document.createElement('select');
+
+  const attr = attributes[i];
+
+  label.textContent = `${attr.charAt(0).toUpperCase() + attr.slice(1, attr.length)}:`;
+
+  input.setAttribute('data-qa', attr);
+  input.setAttribute('required', 'true');
+  input.setAttribute('name', attr);
+  input.setAttribute('type', 'text');
+
+  if (attr === 'age' || attr === 'salary') {
+    input.setAttribute('type', 'number');
+  }
+
+  select.setAttribute('name', 'office');
+  select.setAttribute('data-qa', 'office');
+
+  label.appendChild(input);
+
+  if (attr === 'office') {
+    const labelSelect = document.createElement('label');
+
+    labelSelect.textContent = `${attr.charAt(0).toUpperCase() + attr.slice(1, attr.length)}:`;
+
+    for (let j = 0; j < options.length; j++) {
+      const option = document.createElement('option');
+
+      const optionText = options[j];
+
+      option.value = optionText;
+
+      const text =
+        optionText.charAt(0).toUpperCase() + optionText.slice(1, option.length);
```

Comment:
The `slice` method is incorrectly used here. It should be `optionText.slice(1)` instead of `option.slice(1, option.length)` to correctly capitalize the first letter of the option text.